### PR TITLE
Allow for off-chain oracle poc receipts

### PR DESCRIPTION
**Pull request by Vagabond**

### CODE REVIEW

File: include/blockchain_vars.hrl [hunk 1]
```diff
@@ -238,6 +238,9 @@
 %% whether or not to process validation of poc receipt witnesses
 -define(poc_receipt_witness_validation, poc_receipt_witness_validation).
 
+%% for off-chain poc, who should be signing receipts
+-define(poc_oracle_key, poc_oracle_key).
```
Comment (abhay) [inline include/blockchain_vars.hrl:242]: maybe an array so we can swap them as needed?

Comment (Vagabond) [reply include/blockchain_vars.hrl:242]: would you not just update the var, or are you proposing having a set of them?

Comment (abhay) [reply include/blockchain_vars.hrl:242]: proposing having a list of them so we can roll keys if we need to.

Comment (Vagabond) [reply include/blockchain_vars.hrl:242]: we could also use a multisig key and allow 1/M signing?

Comment (abhay) [reply include/blockchain_vars.hrl:242]: > we could also use a multisig key and allow 1/M signing?

not a bad idea but then we'd have to generate the keys in advance. i'm okay with whatever implementation you think is best.

Comment (Vagabond) [reply include/blockchain_vars.hrl:242]: I mean if it's a chain var we can change the "oracle public key" at any time to reflect keys being added/removed from the rotation?

File: rebar.config [hunk 2]
```diff
@@ -40,7 +40,7 @@
     {erlang_stats, ".*", {git, "https://github.com/helium/erlang-stats.git", {branch, "master"}}},
     {cream, ".*", {git, "https://github.com/helium/cream", {branch, "main"}}},
     {vincenty, ".*", {git, "https://github.com/helium/vincenty", {branch, "master"}}},
-    {helium_proto, {git, "https://github.com/helium/proto.git", {branch, "master"}}},
+    {helium_proto, {git, "https://github.com/helium/proto.git", {branch, "adt/off-chain-poc-verification"}}},
```
Comment (jeffgrunewald) [inline rebar.config:43]: i think this has been merged so this needs updated which will probably fix the rebase conflict

File: src/transactions/v2/blockchain_txn_poc_receipts_v2.erl [hunk 3]
```diff
@@ -156,20 +156,46 @@ is_valid(Txn, Chain) ->
...
-                            check_is_valid_poc(POCVersion, Txn, Chain)
+            {error, empty_path};
+        false ->
+            case libp2p_crypto:verify(EncodedTxn, Signature, PubKey) of
+                false ->
+                    {error, bad_signature};
+                true ->
+                    case {?get_var(?poc_challenger_type, Ledger), ?get_var(?poc_oracle_key, Ledger)} of
+                        {{ok, oracle}, {ok, OracleKey}} ->
+                            case OracleKey == Challenger of
```
Comment (jeffgrunewald) [inline src/transactions/v2/blockchain_txn_poc_receipts_v2.erl:169]: are we just setting the oracle signing key in the poc receipts coming from the oracle?

**Review by jeffgrunewald — APPROVED** [review]

**Review by vihu — APPROVED** [review]

**Pull request merged**